File: apps/backend/src/chat/session/session.manager.ts
// src/chat/session-manager.ts
interface Session {
  userId: string;
  history: { question: string; answer: string }[];
}

export class SessionManager {
  private sessions: Map<string, Session> = new Map();

  getSession(userId: string): Session {
    if (!this.sessions.has(userId)) {
      this.sessions.set(userId, { userId, history: [] });
    }
    return this.sessions.get(userId)!;
  }

  addMessage(userId: string, question: string, answer: string) {
    const session = this.getSession(userId);
    session.history.push({ question, answer });
  }

  clearSession(userId: string) {
    this.sessions.delete(userId);
  }
}

File: apps/ai-server/rag/session_manager.py
# rag/session_manager.py
class SessionManager:
    def __init__(self):
        self.sessions = {}

    def add_message(self, user_Id, question, answer):
        if user_Id not in self.sessions:
            self.sessions[user_Id] = []
        self.sessions[user_Id].append({"question": question, "answer": answer})

    def get_session(self, user_Id):
        return self.sessions.get(user_Id, [])

    def clear_session(self, user_Id):
        if user_Id in self.sessions:
            self.sessions[user_Id] = []
            
session_manager = SessionManager()
